src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory_Methods.cs

```diff
@@ -206,28 +206,62 @@ private IInvalidOperation CreateInvalidExpressionForHasArgumentsExpression(Bound
 
         private ImmutableArray<IOperation> GetAnonymousObjectCreationInitializers(BoundAnonymousObjectCreationExpression expression)
         {
-            // For error cases, the binder generates only the argument.
+            // For error cases and non-assignment initializers, the binder generates only the argument.
             Debug.Assert(expression.Arguments.Length >= expression.Declarations.Length);
 
+            var properties = expression.Type.GetMembers().OfType<IPropertySymbol>().ToImmutableArray();
+            Debug.Assert(properties.Length == expression.Arguments.Length);
+
             var builder = ArrayBuilder<IOperation>.GetInstance(expression.Arguments.Length);
+            var currentDeclarationIndex = 0;
             for (int i = 0; i < expression.Arguments.Length; i++)
             {
                 IOperation value = Create(expression.Arguments[i]);
-                if (i >= expression.Declarations.Length)
+
+                IOperation target;
+                bool isImplicitAssignment;
+
+                // Synthesize an implicit receiver for property reference being assigned.
+                var instance = new InstanceReferenceExpression(
+                        referenceKind: InstanceReferenceKind.ImplicitAnonymousReceiver,
+                        semanticModel: _semanticModel,
+                        syntax: expression.Syntax,
+                        type: expression.Type,
+                        constantValue: default,
+                        isImplicit: true);
+
+                // Find matching declaration for the current argument.
+                if (currentDeclarationIndex >= expression.Declarations.Length ||
+                    i != expression.Declarations[currentDeclarationIndex].PropertyIndex)
```

Comment:
>i != expression.Declarations[currentDeclarationIndex].PropertyIndex [](start = 20, length = 67)

We have a property symbol that we can compare. Can we do without the  ```PropertyIndex``` after all?